packages/core/core/src/Graph.js

```diff
@@ -326,6 +365,54 @@ export default class Graph<TNode: Node> {
     return graph;
   }
 
+  findAncestor(node: TNode, fn: (node: TNode) => boolean): ?TNode {
+    let res = null;
+    this.traverseAncestors(node, (node, ctx, traversal) => {
+      if (fn(node)) {
+        res = node;
+        traversal.stop();
+      }
+    });
+    return res;
+  }
+
+  findAncestors(node: TNode, fn: (node: TNode) => boolean): Array<TNode> {
+    let res = [];
+    this.traverseAncestors(node, (node, ctx, traversal) => {
+      if (fn(node)) {
+        res.push(node);
+        traversal.skipChildren();
+      }
+    });
+    return res;
+  }
+
+  findDescendant(node: TNode, fn: (node: TNode) => boolean): ?TNode {
+    let res = null;
+    this.traverse((node, ctx, traversal) => {
+      if (fn(node)) {
+        res = node;
+        traversal.stop();
+      }
+    }, node);
+    return res;
+  }
+
+  findDescendants(node: TNode, fn: (node: TNode) => boolean): Array<TNode> {
+    let res = [];
+    this.traverse((node, ctx, traversal) => {
+      if (fn(node)) {
+        res.push(node);
+        traversal.skipChildren();
+      }
+    }, node);
+    return res;
+  }
+
+  findNode(predicate: TNode => boolean): ?TNode {
+    return Array.from(this.nodes.values()).find(predicate);
```

Comment:
we should probably be consistent with `[...iterable]` vs `Array.from(iterable)` 😃